pkg/authenticator/ecrsecret.go

```diff
@@ -0,0 +1,134 @@
+package authenticator
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	varPackagesNamespace = "eksa-packages"
+	varConfigMapName     = "ns-secret-map"
+	varECRTokenName      = "ecr-token"
+)
+
+type ecrSecret struct {
+	clientset    kubernetes.Interface
+	nsReleaseMap map[string]string
+}
+
+var _ Authenticator = (*ecrSecret)(nil)
+
+func NewECRSecret(config *rest.Config) (*ecrSecret, error) {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	cm, err := clientset.CoreV1().ConfigMaps(varPackagesNamespace).
+		Get(context.TODO(), varConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return &ecrSecret{
+		clientset:    clientset,
+		nsReleaseMap: cm.Data,
+	}, nil
+}
+
+func (s *ecrSecret) AuthFilename() string {
```

Comment:
So I wanted to keep this function here only because it is possible if there is a move to private helm chart this library can then use the ecr-token secret to generate an authfile that can then be used by helm driver. This would be an implementation that is unique to this and not in the helm driver library.

Reply:
If you really think it's that important, but I'd say it's a tomorrow problem, and you should deal with it in the way that makes the most sense... tomorrow. :) This being here is likely to confuse others, since it doesn't actually do anything now. It could also color your perceptions of the best solution to the problem tomorrow, when it's actually needed, that is, there might be a better solution to this tomorrow, but you wouldn't see it, because you've already made these decisions today.